### PR TITLE
Add tenant info to IdentityClients.

### PR DESCRIPTION
**Pull request by maliming**
This allows us to specify individual configurations for tenants in `IdentityClients`.

```json
{
    "IdentityClients": {
        "Default": {
            "GrantType": "password",
            "ClientId": "Test_App",
            "ClientSecret": "1q2w3e*",
            "UserName": "host_default_admin",
            "UserPassword": "1q2w3E*",
            "Authority": "https://localhost:44395",
            "Scope": "Test_Scope"
        },
        "Default.Test_Tenant1": {
            "GrantType": "password",
            "ClientId": "Test_App",
            "ClientSecret": "1q2w3e*",
            "UserName": "tenant_default_admin",
            "UserPassword": "1q2w3E*",
            "Authority": "https://localhost:44395",
            "Scope": "Test_Scope"
        },
        "Identity": {
            "GrantType": "password",
            "ClientId": "Test_App",
            "ClientSecret": "1q2w3e*",
            "UserName": "host_identity_admin",
            "UserPassword": "1q2w3E*",
            "Authority": "https://localhost:44395",
            "Scope": "Test_Scope"
        },
        "Identity.f72a344f-651e-49f0-85f6-be260a10e4df": {
            "GrantType": "password",
            "ClientId": "Test_App",
            "ClientSecret": "1q2w3e*",
            "UserName": "tenant_identity_admin",
            "UserPassword": "1q2w3E*",
            "Authority": "https://localhost:44395",
            "Scope": "Test_Scope"
        }
    }
}
```

### CODE REVIEW

**Comment by MeryemeBaddaz** [comment]
hello ! I am currently working on a client management project with .net core ef abp framworke and angular , how can i add profile image ?
any one help me !!

File: framework/src/Volo.Abp.IdentityModel/Volo/Abp/IdentityModel/AbpIdentityClientOptions.cs [hunk 1]
```diff
@@ -8,5 +13,30 @@ public AbpIdentityClientOptions()
...
+        {
+            if (identityClientName.IsNullOrWhiteSpace())
+            {
+                identityClientName = IdentityClientConfigurationDictionary.DefaultName;
+            }
+
+            if (currentTenant.Id.HasValue)
+            {
+                var tenantConfiguration = IdentityClients.FirstOrDefault(x => x.Key == $"{identityClientName}.{currentTenant.Id}");
+                if (tenantConfiguration.Key == null && !currentTenant.Name.IsNullOrWhiteSpace())
```
Comment (hikalkan) [inline framework/src/Volo.Abp.IdentityModel/Volo/Abp/IdentityModel/AbpIdentityClientOptions.cs:27]: What if `tenantConfiguration` is null? FirstOrDefault returns null.

File: framework/src/Volo.Abp.IdentityModel/Volo/Abp/IdentityModel/AbpIdentityClientOptions.cs [hunk 2]
```diff
@@ -8,5 +13,30 @@ public AbpIdentityClientOptions()
...
+                {
+                    tenantConfiguration = IdentityClients.FirstOrDefault(x => x.Key == $"{identityClientName}.{currentTenant.Name}");
+                }
+
+                if (tenantConfiguration.Key != null)
```
Comment (hikalkan) [inline framework/src/Volo.Abp.IdentityModel/Volo/Abp/IdentityModel/AbpIdentityClientOptions.cs:32]: tenantConfiguration can be null.

**Review by hikalkan — CHANGES_REQUESTED** [review]

File: framework/src/Volo.Abp.IdentityModel/Volo/Abp/IdentityModel/AbpIdentityClientOptions.cs (hunk 1 above)
Comment (maliming) [reply framework/src/Volo.Abp.IdentityModel/Volo/Abp/IdentityModel/AbpIdentityClientOptions.cs:27]: The type of `tenantConfiguration `  is` KeyValuePair<string, IdentityClientConfiguration>`, which is `struct`.

**Review by hikalkan — APPROVED** [review]

**Pull request merged**